backend/s3/file.go

```diff
@@ -104,8 +91,8 @@ func (f *File) Location() vfs.Location {
 // method if the target file is also on S3, otherwise uses io.CopyBuffer.
 func (f *File) CopyToFile(file vfs.File) error {
 	// validate seek is at 0,0 before doing copy
-	if err := backend.ValidateCopySeekPosition(f); err != nil {
-		return err
+	if f.cursorPos != 0 {
+		return vfs.CopyToNotPossible
 	}
```

Comment:
We're now able to rely on the cursor position (since we track that now) to determine if it's safe to copy (cursor position must be zero)